src/compiler/transformers/esnext.ts

```diff
@@ -177,7 +177,7 @@ namespace ts {
                         chunkObject.push(createPropertyAssignment(p.name, visitNode(p.initializer, visitor, isExpression)));
                     }
                     else {
-                        chunkObject.push(e as ShorthandPropertyAssignment);
+                        chunkObject.push(visitNode(e as ObjectLiteralElementLike, visitor, isObjectLiteralElementLike));

```

Comment:
You could remove the cast by changing the type of `elements` to `ObjectLiteralElementLike[]`.